### DISCUSSION TITLE
Different Port for birdnetpi.local

### DISCUSSION THREAD

**Discussion by MueJosh** [Nachtzuster/BirdNET-Pi#83] [Q&A] [answered]
Hey,

I'm testing allsky on my Raspberry 5 and want it to run at the same time as BirdNET-Pi.
For now `birdnetpi.local` does not work, it gives me the allsky page. 
Is there a way to manually set the Port of BirdNET-Pi instead of the default Port 80? - so I can reach it with `birdnetpi.local:8X`

**Comment by lloydbayley** [comment, marked as answer]
You'll have to edit (with sudo) the Caddyfile which is located at /etc/caddy/Caddyfile
The original will look like this:
```
http:// {
  root * /home/pi/BirdSongs/Extracted
  file_server browse
  handle /By_Date/* {
    file_server browse
  }
  handle /Charts/* {
    file_server browse
  }
  reverse_proxy /stream localhost:8000
  php_fastcgi unix//run/php/php-fpm.sock
  reverse_proxy /log* localhost:8080
  reverse_proxy /stats* localhost:8501
  reverse_proxy /terminal* localhost:8888
}
```

Insert the port you want at the top, for example, to use 7171, use:

```
http://:7171 {
  root * /home/serveradmin/BirdSongs/Extracted
  file_server browse
  handle /By_Date/* {
    file_server browse
  }
  handle /Charts/* {
    file_server browse
  }
  reverse_proxy /stream localhost:8000
  php_fastcgi unix//run/php/php-fpm.sock
  reverse_proxy /log* localhost:8080
  reverse_proxy /stats* localhost:8501
  reverse_proxy /terminal* localhost:8888
}

```

Be very careful with your syntax as Caddy will complain LOUDLY if you aren't!!

After this, you will need to restart Caddy with

```sudo systemctl restart caddy```

**Reply by MueJosh** [reply]
> You'll have to edit (with sudo) the Caddyfile which is located at /etc/caddy/Caddyfile
> 

I've done this before and it didn't work because I forgot to do the second step...

> 
> After this, you will need to restart Caddy with
> 
> 
> ```sudo systemctl restart caddy```
> 


Thank you so much!

**Reply by lloydbayley** [reply]
Excellent! Glad it worked! :)

**Comment by salado912** [comment]
this took me way to long to figure out BUT you can set it up with both an Internal IP and External IP in the same /etc/caddy/Caddyfile.
top example is internal IP of BirdNet-pi - Leave blank to use your pi 's static LAN address
bottom example is EXTERNAL IP and port you want to use in port fwd, example used was 7171 (only put port#)
____________________________________________________________________________________________________________

http:// {
	root * /home/yourname/BirdSongs/Extracted
	file_server browse
	handle /By_Date/* {
		file_server browse
	}
	handle /Charts/* {
		file_server browse
	}
	basicauth /views.php?view=File* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /Processed* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /scripts* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /stream {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /phpsysinfo* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /terminal* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	reverse_proxy /stream localhost:8000
	php_fastcgi unix//run/php/php-fpm.sock
	reverse_proxy /log* localhost:8080
	reverse_proxy /stats* localhost:8501
	reverse_proxy /terminal* localhost:8888
}


http://:7171 {
  	root * /home/yourname/BirdSongs/Extracted
	file_server browse
	handle /By_Date/* {
		file_server browse
	}
	handle /Charts/* {
		file_server browse
	}
	basicauth /views.php?view=File* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /Processed* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /scripts* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /stream {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /phpsysinfo* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	basicauth /terminal* {
		birdnet $2a$14$gx0k7j8vasBRQMp2Si51K
	}
	reverse_proxy /stream localhost:8000
	php_fastcgi unix//run/php/php-fpm.sock
	reverse_proxy /log* localhost:8080
	reverse_proxy /stats* localhost:8501
	reverse_proxy /terminal* localhost:8888
}

**Comment by twostar01** [comment]
Funny enough, I did the same thing but found it easier to change the port on the allsky implementation. Thanks for asking here though because I'm getting ready to do a rebuild on the box and I want to implement a better home page for the box that links off to the different services available.

Out of curiousity, do you have any other projects you're running on your pi? I'm trying to make this a great remote nature observation box.

**Reply by MueJosh** [reply]
> Out of curiousity, do you have any other projects you're running on your pi? I'm trying to make this a great remote nature observation box.

I recently moved house and have now separated everything again.
That means: one Raspberry Pi Zero for this BirdNET Fork, one Raspi 4 4Gb for [BattyBirdNET-Pi](https://github.com/rdz-oss/BattyBirdNET-Pi), and one Raspi 5 8Gb (with SSD +PoE) for [Allsky](https://github.com/AllskyTeam/allsky). 

Unfortunately I don't have time to set everything up to run on just one Raspi at the moment, but I'll try again sometime.

Weather data would also be interesting to draw conclusions about the call frequency of birds and bats.
Rain and cloud detection would be easy to implement with Allsky. 
Temperature+humidity (and wind) would be very interesting. 

[A completely different topic would be the localization of calls via tdoa localization. This would require several stations (at least 3) and you would need a GPS module for the exact time. 
If a script writes the data into a single CSV together with the other data... a dream. 
I would like to have more time :D ]

**Reply by twostar01** [reply]
Funny enough, I've got a weather station right next to the pi and I'm collecting the data via Weewx on another box so it's available, just need to figure out how to tie the data streams together. I've played with cloud detection in allsky but got poor results. I've put that to the side while I work on getting a new hardware build out with solar power and local battery storage worked out. I'd like to remove the extension cord going across the lawn! 

Now I've got to look at the batty fork and make more hard decisions. Thanks for the info!
